buildtools/src/main/java/org/apache/pulsar/tests/ThreadLeakDetectorListener.java

```diff
@@ -201,10 +213,32 @@ private static boolean shouldSkipThread(Thread thread) {
             if (threadName.equals("Grizzly-HttpSession-Expirer")) {
                 return true;
             }
+            // Testcontainers AbstractWaitStrategy.EXECUTOR
+            if (threadName.startsWith("testcontainers-wait-")) {
+                return true;
+            }
+        }
+        Runnable target = extractRunnableTarget(thread);
+        if (target != null) {
+            String targetClassName = target.getClass().getName();
+            // ignore threads that contain a Runnable class under org.testcontainers package
+            if (targetClassName.startsWith("org.testcontainers.")) {
+                return true;
+            }
         }
         return false;
     }
 
+    private static Runnable extractRunnableTarget(Thread thread) {
```

Comment:
NIT: maybe add a comment on why we have to use reflection here.

Reply:
Done